src/txmempool.h

```diff
@@ -605,6 +605,9 @@ class CTxMemPool
     /** Expire all transaction (and their dependencies) in the mempool older than time. Return the number of removed transactions. */
     int Expire(int64_t time);
 
+    /** Returns false if the transaction is in the mempool and not within the chain limit specified. */
+    bool CoinWithinLimit(const uint256& txid, size_t chainLimit) const;
```

Comment:
Should we call this something other than **Coin**WithinLimit?  Seems like coins are not something that the mempool is aware of.